### PR TITLE
Heading Typescript migration

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/2861766393/pulses/3369806530

### CODE REVIEW

File: src/components/EditableHeading/EditableHeading.jsx [hunk 1]
```diff
@@ -200,8 +198,8 @@ const EditableHeading = props => {
   );
 };
 
-EditableHeading.types = TYPES;
-EditableHeading.sizes = SIZES;
+EditableHeading.types = Heading.types;
```
Comment (hadasfa) [inline src/components/EditableHeading/EditableHeading.jsx:201]: Let's switch to object assign format as we did in other components

File: src/components/Heading/Heading.tsx [hunk 2]
```diff
@@ -1,41 +1,61 @@
...
   style,
   tooltipPosition,
-  highlightTerm,
-  suggestEditOnHover,
-  brandFont,
-  nonEllipsisTooltip // tooltip to show when no overflow
+  highlightTerm = null,
+  suggestEditOnHover = false,
+  brandFont = false,
+  nonEllipsisTooltip = null // tooltip to show when no overflow
```
Comment (hadasfa) [inline src/components/Heading/Heading.tsx:48]: let's move this comment to be on the interface declaration for be appeared on storybook

File: src/components/TextWithHighlight/TextWithHighlight.tsx [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+  key: number,
+  shouldHighlight: boolean,
+  wrappingTextTag = "em",
+  wrappingElementClassName: string
+) => {
   const WrappingElement = wrappingTextTag;
   if (shouldHighlight) {
     return (
+      // @ts-ignore variable-component
       <WrappingElement className={cx("highlight-text", wrappingElementClassName)} key={key}>
```
Comment (hadasfa) [inline src/components/TextWithHighlight/TextWithHighlight.tsx:21]: I am thinking there is a typescript way to implement this without ts ignore. let's check?

File: webpack/published-js-components.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ const publishedJSComponents = {
   Box: "components/Box/Box.jsx",
   AccordionItem: "components/Accordion/AccordionItem/AccordionItem.jsx",
   Accordion: "components/Accordion/Accordion/Accordion.jsx",
-  TextWithHighlight: "components/TextWithHighlight/TextWithHighlight.jsx",
+  TextWithHighlight: "components/TextWithHighlight/TextWithHighlight.tsx",
```
Comment (hadasfa) [inline webpack/published-js-components.js:7]: if its converted please move it to published-ts-components without the .tsx extension

**Review by hadasfa — CHANGES_REQUESTED** [review]
please read comments

File: src/components/EditableHeading/EditableHeading.jsx (hunk 1 above)
Comment (SergeyRoyt) [reply src/components/EditableHeading/EditableHeading.jsx:201]: This component haven't been converted to TS yet - will do it while converting

File: src/components/Heading/Heading.tsx (hunk 2 above)
Comment (SergeyRoyt) [reply src/components/Heading/Heading.tsx:48]: Done

File: src/components/TextWithHighlight/TextWithHighlight.tsx (hunk 3 above)
Comment (SergeyRoyt) [reply src/components/TextWithHighlight/TextWithHighlight.tsx:21]: I'll check

File: webpack/published-js-components.js (hunk 4 above)
Comment (SergeyRoyt) [reply webpack/published-js-components.js:7]: Fixed

File: src/components/TextWithHighlight/TextWithHighlight.tsx (hunk 3 above)
Comment (SergeyRoyt) [reply src/components/TextWithHighlight/TextWithHighlight.tsx]: Fixed 👍

**Review by hadasfa — APPROVED** [review]

**Pull request merged**